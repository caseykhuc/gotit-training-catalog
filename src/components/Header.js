import React from 'react';
import { Button, ButtonGroup } from 'react-bootstrap';
import { connect } from 'react-redux';
import { showModal } from '../actions/modal';
import { signoutUser } from '../actions/user';
import modalKeys from '../constants/modelKeys';

const Header = ({ isSignedIn, showModal, signoutUser }) => (
  <div
    style={{
      display: 'flex',
      alignItems: 'center',
      justifyContent: 'space-between',
    }}
  >
    <h1>Catalog Header</h1>
    {isSignedIn
      ? (
        <Button
          onClick={signoutUser}
        >
          Sign Out
        </Button>
      )
      : (
        <ButtonGroup>
          <Button
            onClick={() => showModal(modalKeys.REGISTER_MODAL)}
          >
            Register
          </Button>
          <Button onClick={() => showModal(modalKeys.SIGNIN_MODAL)}>
            Sign In
          </Button>
        </ButtonGroup>
      )}
  </div>
)


export default connect(null, { showModal, signoutUser })(Header);
