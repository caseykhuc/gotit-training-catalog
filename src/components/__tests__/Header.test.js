
import React from 'react';
import { shallow } from 'enzyme';
import { Header } from '../Header';
import modalKeys from '../../constants/modelKeys';

describe('component/Header', () => {
  let props;
  let wrapper;
  let authBtn;
  let addBtn;
  let registerBtn;

  const update = () => {
    wrapper.update();
    authBtn = wrapper.find('.auth-btn');
    addBtn = wrapper.find('.add-btn');
    registerBtn = wrapper.find('.register-btn');
  };
  const setup = () => {
    wrapper = shallow(<Header {...props} />);
    update();
  };
  beforeEach(() => {
    props = {
      isSignedIn: false,
      showModal: jest.fn(),
      signoutUser: jest.fn(),
    }
  });
  it('should render correctly', () => {
    setup();
    expect(wrapper).toMatchSnapshot();
  });
  it('should render SignIn button correctly', () => {
    setup();
    expect(authBtn.text()).toBe('Sign In');
    expect(registerBtn).toHaveLength(1);
    registerBtn.simulate('click');
    expect(props.showModal).toHaveBeenCalledWith(modalKeys.REGISTER_MODAL);

    authBtn.simulate('click');
    expect(props.showModal).toHaveBeenCalledWith(modalKeys.SIGNIN_MODAL);
  });
  it('should render SignOut button correctly', () => {
    props.isSignedIn = true;
    setup();
    expect(authBtn.text()).toBe('Sign Out');
    expect(addBtn).toHaveLength(1);
    addBtn.simulate('click');
    expect(props.showModal).toHaveBeenCalledWith(modalKeys.ADD_ITEM_MODAL);

    authBtn.simulate('click');
    expect(props.signoutUser).toHaveBeenCalled();
  })
})
