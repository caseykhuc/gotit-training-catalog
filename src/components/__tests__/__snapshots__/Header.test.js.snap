// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component/Header should render correctly 1`] = `ShallowWrapper {}`;
